README.md

```diff
@@ -1,51 +1,66 @@
-# loopback4-extension-grpc
+# gRPC Extension for LoopBack 4
 
 [![Join the chat at https://gitter.im/strongloop/loopback4-extension-grpc](https://badges.gitter.im/strongloop/loopback4-extension-grpc.svg)](https://gitter.im/strongloop/loopback4-extension-grpc?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
-gRPC Extension for LoopBack 4
-
-## Summary
-A gRPC extension for LoopBack Next.
-
 ## Overview
-TODO.
+The `@loopback/grpc` component enables LoopBack 4 as a Grpc Server. Also it provides with a @grpc decorator to define your RPC Method implementations from your Application Controllers.
 
 ## Installation
-Get started by either downloading this project or cloning it as follows:
+Install the `@loopback/grpc` component in your LoopBack 4 Application.
 
 ```sh
-$ git clone https://github.com/strongloop/loopback4-extension-grpc.git
-$ cd loopback4-extension-grpc && npm install
+$ npm install --save @loopback/grpc
 ```
 
-## Basic Usage
-TODO 
-
-### Components
-TODO.
-
-### Controllers
-TODO.
+## Component Configuration
+```js
+import {Application} from '@loopback/core';
+import {GrpcComponent, GrpcConfig} from '@loopback/grpc';
+import {MyGreeter} from './MyGreeter';
 
-### Decorators
-TODO.
+const config: GrpcConfig = {
+    proto: './file.proto',
+    package: 'myawesomepackage'
+}
 
-### Mixins
-TODO. 
+const app = new Application({
+    components: [GrpcComponent],
+    grpc: config
+});
 
-### Providers
-TODO. 
+app.controller(MyGreeter});
 
-### Repositories
-TODO.
+await app.start();
+```
 
-### Project Layout
+## Grpc Decorator
+The `@loopback/grpc` component provides you with a handy decorator to implement GRPC Methods.
+
+```js
+import {grpc} from '@loopback/grpc';
+// You create the following types according your own proto.file
+import {Greeter, HelloRequest, HelloReply} from './types';
```

Comment:
Are these generated gRPC interfaces/classes?

Reply:
Look this awesome tool that @VMois found https://geotho.github.io/protobuf-to-typescript/

Reply:
It does convert great the messages but it looks service and rpc methods are not so good, but yeah the idea is to autogenerate those from the proto file

````txt
package greeterpackage;
syntax = "proto3";
 
service Greeter {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
````